avocado/utils/process.py

```diff
@@ -750,7 +756,13 @@ def send_signal(self, sig):
         :param sig: Signal to send.
         """
         self._init_subprocess()
-        self._popen.send_signal(sig)
+        if self.is_running_as_sudo():
+            for child in get_children_pids(self.get_pid()):
+                child_pid = child.decode('UTF-8')
```

Comment:
`get_children_pid` is supposed to return a `list of int`.  If it is, there's no need for this line here.  If it's not, it needs to be fixed.